tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -973,6 +972,33 @@ duckdb::pyarrow::RecordBatchReader DuckDBPyConnection::FetchRecordBatchReader(co
 	}
 	return result->FetchRecordBatchReader(chunk_size);
 }
+
+static bool IsPolarsDataFrame(py::object entry) {
+	auto polars_df_type = pybind11::module_::import("polars").attr("DataFrame");
+	return py::isinstance(entry, polars_df_type);
```

Comment:
Once you've fixed the check_ method on the class, you can just use that here with the template variant of this function